packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -187,6 +193,8 @@ const SUSPENSE_START_DATA = '$';
 const SUSPENSE_END_DATA = '/$';
 const SUSPENSE_PENDING_START_DATA = '$?';
 const SUSPENSE_FALLBACK_START_DATA = '$!';
+const FORM_STATE_IS_MATCHING = '!F';
```

Comment:
Nit: I understand you're using this to mean `NOT form state` but the pattern that the other signaling comments use is:

TAG + STATE

So you can just check the first character to see if it matches the type of thing and then the second character is the state of that thing.

We should probably stick to that pattern. In theory if we did like `switch(data[0])` it could matter.